hail/python/hail/ir/table_reader.py

```diff
@@ -0,0 +1,56 @@
+import abc
+import json
+
+from hail.typecheck import *
+from hail.utils.java import escape_str
+
+class TableReader(object):
+    @abc.abstractmethod
+    def render(self, r):
+        pass
+
+    @abc.abstractmethod
+    def __eq__(self, other):
+        pass
+
+
+class TableNativeReader(TableReader):
+    @typecheck_method(path=str)
+    def __init__(self, path):
+        self.path = path
+
+    def render(self, r):
+        reader = {'name': 'TableNativeReader',
+                  'path': self.path}
+        return escape_str(json.dumps(reader))
+
+    def __eq__(self, other):
+        return isinstance(other, TableNativeReader) and \
+               other.path == self.path
+
+class TextTableReader(TableReader):
+    def __init__(self, paths, min_partitions, types, comment,
+                 delimiter, missing, no_header, impute, quote,
+                 skip_blank_lines, force_bgz):
+        self.config = {
+            'files': paths,
+            'typeMapStr': {f: t._parsable_string() for f, t in types.items()},
+            'comment': comment,
+            'separator': delimiter,
+            'missing': missing,
+            'noHeader': no_header,
+            'impute': impute,
+            'nPartitions': None,
```

Comment:
min_partitions isn't used in the config at all? Is it supposed to go here?

Reply:
Oops, good catch, it was supposed to go here.